### PR TITLE
Add ESLint rules for telephone numbers

### PR DESCRIPTION
**Pull request by bkjohnson**
## Description

Part of https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/864

This adds two new custom ESLint rules:

### prefer-telephone-component

This is an autofixable ESLint rule which will trigger on the pattern `<a href="tel:..."`. When run with `--fix`, it will replace that with `<va-telephone>`. It will also:

- replace the `href` attribute with `contact
- strip all non-digits out of the contact number
- remove child content since the component doesn't allow it
- remove the `aria-label` if present since the component will add its own to the link

### telephone-contact-3-or-10-digits

This is a rule added as part of this PR to try to prevent the first rule from applying a fix that results in the `<va-telephone>` component having a bad `contact` prop, i.e. one that doesn't have 3 or 10 digits. It might flag on code that results from the autofixable rule above, but also on [code like this](https://github.com/department-of-veterans-affairs/vets-website/blob/787b2486cf316ec1c5b4535866ee2d732e08bf26/src/applications/financial-status-report/components/AvailableDebts.jsx#L106).

This rule will try to remove any non-digit characters (i.e. hyphens) from the `contact` prop that it can, since they aren't digits and extend the length beyond 10.

This specific constraint is mentioned on [Storybook](https://design.va.gov/storybook/?path=/docs/components-va-telephone--default):

Screenshot of the telephone component's storybook page with the contact prop's description highlighted. It reads "3 or 10 digit string representing the contact number"

As well as on the [component's guidance page](https://design.va.gov/components/telephone#code-usage):

Screenshot of the component's page on design.va.gov. Under the "How to use va-telephone" heading the bullet point "Add a 3 or 10 digit phone number to the component to have it formatted correctly for usage in a page." is highlighted. Below that under "Code usage", in the properties table the description for `contact` is once again highlighted.

## Testing done

- Local unit testing with ESLint's `RuleTester`
- Testing in `vets-website`

## Screenshots

Local testing in `vets-website` using:

```diff
diff --git a/package.json b/package.json
index b8e8ea8e0c..217bac41b7 100644
--- a/package.json
+++ b/package.json
@@ -124,7 +124,7 @@
     "@babel/traverse": "^7.17.3",
     "@bahmutov/cypress-esbuild-preprocessor": "^2.1.2",
     "@cypress/code-coverage": "^3.10.0",
-    "@department-of-veterans-affairs/eslint-plugin": "^1.2.2",
+    "@department-of-veterans-affairs/eslint-plugin": "link:../veteran-facing-services-tools/packages/eslint-plugin",
     "@department-of-veterans-affairs/generator-vets-website": "^3.8.0",
     "@octokit/rest": "^18.11.0",
     "@pact-foundation/pact": "^9.16.4",
```

A screenshot of vim with a file opened in the vets-website repo. There is an anchor link with a tel href, with an ESLint warning saying "The va-telephone web component should be used to display phone numbers".

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Changes have been tested in vets-website
- [ ] Changes have been tested in IE11, if applicable
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: packages/eslint-plugin/lib/rules/prefer-telephone-component.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+        const anchorNode = node.openingElement;
+
+        const hrefProp = getProp(anchorNode.attributes, 'href');
+        const hrefValue = getLiteralPropValue(hrefProp);
+
+        if (hrefValue?.startsWith('tel:')) {
+          const contact = hrefValue
+            .substring('tel:'.length)
+            .replace(/[^\d]/g, '');
```
Comment (bkjohnson) [inline packages/eslint-plugin/lib/rules/prefer-telephone-component.js:28]: I decided against having this rule do any kind of length checking here. The [ESLint docs on applying fixes](https://eslint.org/docs/latest/developer-guide/working-with-rules#applying-fixes) say under best practices:

> Make fixes as small as possible. Fixes that are unnecessarily large could conflict with other fixes, and prevent them from being applied.

I didn't want to have a single rule rewrite code from an `<a>` to a `<va-telephone>` _and_ potentially truncate a phone number by removing some digits. I also didn't want to have this rule ignore `tel` links, so I wrote the second rule to take care of notifying developers when the `contact` prop doesn't have the right number of digits.

**Comment by jamigibbs** [comment]
This will still complain for international numbers. Are we ok with that being an exception? Example at `src/applications/toe-benefits/components/underReview.jsx`

Screen Shot 2022-08-25 at 12 40 32 PM

**Comment by jamigibbs** [comment]
I'm not sure if this is possible but will it allow us to control the spacing output? For example, prettier will complain that a space is needed before the end-slash and also sometimes there is a double space before `contact`:

Screen Shot 2022-08-25 at 12 43 58 PM

Screen Shot 2022-08-25 at 12 46 23 PM

**Comment by bkjohnson** [comment]
I'm ok with that. That number also has two leading 0s, and in our documentation we say to use the `international` prop for international numbers. 

Screenshot of the telephone page on storybook with the "internation" prop highlighted in the args table. The description reads "Indicates if this is a number meant to be called from outside the US. Prepends a "+1" to the formatted number."

I was trying to avoid having this work creep into trying to detect if a number _should_ be marked as `international` or not since the original goal was to get `<a href="tel"` links to be a `<va-telephone>`.

Do you think I should add a third rule which looks for 11 digit contact numbers beginning with `1`, or 13 digit contact numbers beginning with `001` and removes those and adds an `international` prefix?

**Comment by jamigibbs** [comment]
> Do you think I should add a third rule which looks for 11 digit contact numbers beginning with 1, or 13 digit contact numbers beginning with 001 and removes those and adds an international prop?

I can definitely see a bit of scope creep there and I'm fine with keeping this just for domestic numbers. But the warning message is confusing when you read it next to an international number. It makes it seem like there's a bug in the linter. 

Would things become too verbose if the warning text is just updated to make mention of internationl numbers being an exception or to use the international prop? Or even just "The <va-telephone> contact prop should only contain digits and be 3 or 10 digits long _for domestic numbers_."

**Comment by bkjohnson** [comment]
What do you think about this for a message?

> The <va-telephone> contact prop should only contain digits and be 3 or 10 digits long. Consider the international prop if you want the number to begin with a 1.

**Comment by jamigibbs** [comment]
> What do you think about this for a message?
> 
> > The  contact prop should only contain digits and be 3 or 10 digits long. Consider the international prop if you want the number to begin with a 1.

That's much clearer to me! 👍

**Review by jamigibbs — APPROVED** [review]

**Review by theodur — APPROVED** [review]

**Pull request merged**